sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -912,6 +915,20 @@ case class MapObjects private(
 
     val (initCollection, addElement, getResult): (String, String => String, String) =
       customCollectionCls match {
+        case Some(cls) if classOf[WrappedArray[_]].isAssignableFrom(cls) =>
+          // Scala WrappedArray
+          val getBuilder = s"${cls.getName}$$.MODULE$$.newBuilder()"
+          val builder = ctx.freshName("collectionBuilder")
+          (
+            s"""
+               ${classOf[Builder[_, _]].getName} $builder = $getBuilder;
+               $builder.sizeHint($dataLength);
+             """,
+            (genValue: String) => s"$builder.$$plus$$eq($genValue);",
+            s"(${cls.getName}) ${classOf[WrappedArray[_]].getName}$$." +
+              s"MODULE$$.make(((${classOf[ArrayBuffer[_]].getName})$builder" +
```

Comment:
`$builder.result()` is a `ArrayBuffer`?

Reply:
Isn't `WrappedArray` `HasNewBuilder[T, WrappedArray[T]]`, so `result` should return a `WrappedArray`?

Reply:
> $builder.result() is a ArrayBuffer?

Yes.

Reply:
What do you mean by `HasNewBuilder`? I can't find it.

Reply:
https://www.scala-lang.org/api/2.12.8/scala/collection/mutable/WrappedArray.html and https://www.scala-lang.org/api/2.12.8/scala/collection/generic/HasNewBuilder.html

Reply:
I see. I believe it's the difference between the `WrappedArray` instance and the `WrappedArray` object. The instance one is inherited from `HasNewBuilder[T, WrappedArray[T]]` but the object is not.

Reply:
Oh, I see. Object `WrappedArray`'s `newBuilder` is `def newBuilder[A]: Builder[A, IndexedSeq[A]]`.

Reply:
Then I think `IndexedSeq` is safer?

Reply:
Sounds reasonable.

Reply:
updated, thanks!